src/Sulu/Component/Content/Document/Query/QueryBuilder.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Content\Document\Query;
+
+use Doctrine\ODM\PHPCR\Query\Builder\QueryBuilder as BaseQueryBuilder;
+use Sulu\Component\Content\Document\Query\Builder\Where;
+
+/**
+ * Sulu version of the PHPCR-ODM query builder.
+ *
+ * Introduces the methods which apply to Sulu Structure objects.
+ *
+ * NOTE: The PHPCR-ODM query builder is a "fluid" query builder which supports,
+ *       to a degree, IDE completetion. For this reason it is not easy to
+ *       extend. We have therefore needed to extend classes in order that they
+ *       instantiate objects which inturn extend our factories.
+ *
+ *       This is the reason that there are lots of seemingly pointless classes
+ *       in the Builder\\ directory.
+ */
+class QueryBuilder extends BaseQueryBuilder
+{
+    /**
+     * @var array
+     */
+    private $structureMap;
+
+    /**
+     * Assign a structure to a document alias.
+     *
+     * This allows you to use structure fields in criteria as follows:
+     *
+     * ````
+     * $qb->from()->document('foo', 'p');
+     * $qb->useStructure('p', 'overview');
+     * $qb->where()->eq()->structureField('p.field1')->literal('barbar');
+     * ````
+     *
+     * @param string $documentAlias Document alias to register structure to
+     * @param string $structureName Name of the structure to register.
+     *
+     * @throws \InvalidArgumentException If a document with the given alias is already registered.
+     *
+     * @return self
+     */
+    public function useStructure($documentAlias, $structureName)
```

Comment:
have reversed and the `documentAlias` is optional and can be omitted if only one Document is invovled in the query.